### DISCUSSION TITLE
unverified username even though my server is in offline mode

### DISCUSSION THREAD

**Discussion by Ghir8** [PrismarineJS/mineflayer#2488] [Q&A]
Hi everyone,
this is a tricky situation:
I have a bot to AFK in my server that I have with my friends, before the 1.18 update it worked just fine, but now I can't login.
I'm working with mineflayer version 4.0.0 and a server 1.18.1 with fabric-loader version 12.12

Here's the code:
```
const mineflayer = require('mineflayer')
var v = require('vec3')

const options = {
  host: 'IP_REMOTE_SERVER',
  port: 25565,
  username: 'USR_BOT'
}

const bot = mineflayer.createBot(options)

bot.on('kicked', console.log)
bot.on('error', console.log)

const login = () => {
  ...
}
```

Now, the mineflayer's error is ` {"translate":"multiplayer.disconnect.unverified_username"} false ` and the server's log are:
```
16.02 15:32:04 [Disconnect] User /MY_HOME_IP:52651 has disconnected, reason: Failed to verify username!
16.02 15:32:04 [Server] User Authenticator #14/ERROR Username 'USR_BOT' tried to join with an invalid session
16.02 15:32:04 [Server] User Authenticator #14/INFO Disconnecting /MY_HOME_IP:52651: Failed to verify username!
```

Final note, the server is in offline mode (`online-mode=false`), but I've enabled the whitelist and the bot is registered in it.

**Comment by amoraschi** [comment]
Did you restart the server after applying the changes? Also from 4.0.0 I think the default login method is Microsoft, so you might want to set it to mojang

**Reply by Ghir8** [reply]
I tied restarting the server and forcing the auth method but keeps rejecting the connection.

**Reply by Ghir8** [reply]
but, with the online mode deactivated in the server, I thought that it wasn't needed the auth parameter in the createBot method...

**Reply by Ghir8** [reply]
Interesting development:
the bot doesn't have an official Minecraft account (it is mostly for AFK and playing around with the code), I tried to log in with my official microsoft account and it worked! 
But I do not understand, with the `online-mode=false` setting in the server.preferences file the server shouldn't go online for the authentication, or I'm missing something?
